src/backend.ts

```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  SteamClient.System.UI.RegisterForSystemKeyEvents((e) => {
+    // Have not found any race condition issues with this approach since the key event fires long before focus change
+    validKeyEvent = e;
+  });
+
   const { unregister: unregisterFocusChangeEvents } =
     SteamClient.System.UI.RegisterForFocusChangeEvents(
       throttle(async (fce: FocusChangeEvent) => {
         // don't try anything while an application is launching or it could pause it midlaunch
         if (appIsStartingUp) return;
         // skip if we already got such an event before
-        if (fce.focusedApp.pid === lastPid) return;
+        if (
+          fce.focusedApp.pid === lastPid &&
+          fce.focusedApp.appid === lastAppid
+        )
+          return;
         lastPid = fce.focusedApp.pid;
+        lastAppid = fce.focusedApp.appid;
+        // Only pause if the user has enabled it
+        if (
+          fce.focusedApp.appid === 769 &&
+          (await loadSettings()).overlayPause &&
+          fce.focusedApp.strExeName !== "steamwebhelper"
+        ) {
+          // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
+          if (validKeyEvent?.eKey === 0) {
+            validKeyEvent = null;
+            const appid = await appid_from_pid(fce.focusedApp.pid);
+            if (appid) {
+              const appMD = await getAppMetaData(Number(appid));
+              appMD.is_paused = await is_paused(appMD.instanceid);
+              if (appMD.sticky_state) {
+                return;
+              }
+              if (!appMD.is_paused) {
+                appMD.is_paused = await pause(appMD.instanceid);
+              }
+              return;
```

Comment:
Are these returns intentional ? Because they kinda shortcircuit the "Pause on focus loss" feature when multiple apps are running.

Ah I guess the problem would be that the app might be unpaused since it is the main one.

Reply:
I ran into an issue where apps would not unpause themselves if I didn't return sooner, I believe that's why I added the short circuit

Reply:
Also the later logic would have the app unpause immediately after the pause is triggered, yes.